source/auth_n.rst

```diff
@@ -4915,24 +4875,33 @@ Challenging During Factor Creation
 
   For this example, we will use an SMS challenge. Challenging a Google Authenticator Factor during creation is not feasible because the user has to add the factor to their application before they can get a code.
 
-To send a challenge at the same time as you create the phone Factor, you need to POST to the Account's ``/factors`` endpoint with the additional ``?challenge=true`` parameter included. Then you must also add the ``challenge`` into the body of the JSON.
+.. only:: rest or csharp or nodejs or php or python or vbnet
 
-.. code-block:: http
+  To send a challenge at the same time as you create the SMS Factor, you need to POST to the Account's ``/factors`` endpoint with the additional ``?challenge=true`` parameter included. Then you must also add the ``challenge`` into the body of the JSON.
 
-  POST /v1/accounts/5IvkjoqcYNe3TYMExample/factors?challenge=true HTTP/1.1
-  Host: api.stormpath.com
-  Authorization: Basic MlpG...
-  Content-Type: application/json
-
-  {
-    "type":"sms",
-    "phone": {
-      "number": "2675555555"
-    },
-    "challenge": {
-      "message": "Welcome to the Example! Your authorization code is ${code}"
+  .. code-block:: http
+
+    POST /v1/accounts/5IvkjoqcYNe3TYMExample/factors?challenge=true HTTP/1.1
+    Host: api.stormpath.com
+    Authorization: Basic MlpG...
+    Content-Type: application/json
+
+    {
+      "type":"sms",
+      "phone": {
+        "number": "2675555555"
+      },
+      "challenge": {
+        "message": "Welcome to the Example! Your authorization code is ${code}"
+      }
     }
-  }
+
+.. only:: java
+
+  To create a challenge at the same time as you create the SMS Factor, you need to call ``account.createFactor((Factors.SMS.newCreateRequestFor(factor).createChallenge()).build())``:
```

Comment:
@mrafiei this looks weird. You have code in one line followed by a Java block showing more code. There seems to be some text missing to properly identify what that last code is for.